docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,871 @@
+openstack security group rule create --description "VXLAN from master" --protocol udp --dst-port 4789 --remote-group "$INFRA_ID-master" "$INFRA_ID-worker"
+openstack security group rule create --description "Geneve" --protocol udp --dst-port 6081 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "Geneve from master" --protocol udp --dst-port 6081 --remote-group "$INFRA_ID-master" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress internal (TCP)" --protocol tcp --dst-port 9000:9999 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress internal from master (TCP)" --protocol tcp --dst-port 9000:9999 --remote-group "$INFRA_ID-master" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress internal (UDP)" --protocol udp --dst-port 9000:9999 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress internal from master (UDP)" --protocol udp --dst-port 9000:9999 --remote-group "$INFRA_ID-master" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress kubelet secure" --protocol tcp --dst-port 10250 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress kubelet secure from master" --protocol tcp --dst-port 10250 --remote-group "$INFRA_ID-master" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress services (TCP)" --protocol tcp --dst-port 30000:32767 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "worker ingress services (UDP)" --protocol udp --dst-port 30000:32767 --remote-group "$INFRA_ID-worker" "$INFRA_ID-worker"
+openstack security group rule create --description "VRRP" --protocol vrrp --remote-ip 192.0.2.0/24 "$INFRA_ID-worker"
+```
+
+
+### Router
+
+The outside connectivity will be provided by floating IP addresses. Your subnet needs to have a router set up for this to work.
+
+```sh
+$ openstack router create "$INFRA_ID-router"
+$ openstack router set --external-gateway <external> "$INFRA_ID-router"
+$ openstack router add subnet "$INFRA_ID-router" "$INFRA_ID-subnet"
+```
+
+**NOTE**: Pass in the same external network name (`external` in the `router create` command above) that you used in the install config and that you used in `openstack floating ip create`.
+
+
+### Public API and Ingress Access
+
+To provide access to the OpenShift cluster, we will need to create two ports, attach our API and Ingress Floating IPs and publish the appropriate DNS records.
+
+**IMPORTANT:** these ports will need to have specific fixed (i.e. private) IP addresses. They have to be the fifth and seventh addresses from the subnet range. For example:
```

Comment:
Why does `fixed == private` here? (I might be wrong but that's my interpretation of "id est")

Reply:
In OpenStack terminology, the private IP address (only addressable within the subnet the server/port is) is a `Fixed IP`.

In most classic OpenStack deployments (where you need routers and floating IPs to provide connectivity beyond the subnet) fixed == private and floating == public.

But: there are deployments (the key term is generally "provider networks") where you don't need floating IPs and any server you create will have its fixed IP also accessible from the outside.